vignettes/datatable-faq.Rmd

```diff
@@ -460,6 +460,44 @@ Happily, an internet search for "How does R method dispatch work" (at the time o
 
 However, features like basic S3 dispatch (pasting the function name together with the class name) is why some R folk love R. It's so simple. No complicated registration or signature is required. There isn't much needed to learn. To create the `merge` method for data.table all that was required, literally, was to merely create a function called `merge.data.table`.
 
+## Why do `T` and `F` behave differently from `TRUE` and `FALSE` in some `data.table` queries?
+
+Using `T` and `F` as abbreviations for `TRUE` and `FALSE` in `data.table` can lead to unexpected behavior. This is because `T` and `F` are global variables that can be redefined, which causes them to be treated as variable names rather than logical constants. This issue does not occur with `TRUE` and `FALSE`. Avoiding `T` and `F` is advice for using R generally, but it shows up in data.table in some perhaps surprising ways, for example:
```

Comment:
> Using `T` and `F` as abbreviations for `TRUE` and `FALSE` in `data.table` can lead to unexpected behavior. This is because `T` and `F` are global variables that can be redefined, which causes them to be treated as variable names rather than logical constants. This issue does not occur with `TRUE` and `FALSE`.

Using `T` and `F` as abbreviations for `TRUE` and `FALSE` in `data.table` can lead to unexpected behavior as they are global variables that can be redefined, and are thus treated as variable names rather than reserved logical constants.

(Doesn't have to be what I rephrased above, but I would suggest something slightly less repetitive, and I don't think we need to explicitly mention the somewhat obvious resolution that the issue does not occur when using `TRUE`/`FALSE` instead)

> Avoiding `T` and `F` is advice for using R generally, but it shows up in data.table in some perhaps surprising ways, for example

`data.table`